Source: https://git.alpinelinux.org/aports/tree/community/s6-overlay/execline-compat-fix.patch?h=3.14-stable

Most recent version of execline (2.8.0.0) changed behaviour of 'backtick'
command. When Alpine execline package was updated to this version in
February it broke some execline scripts in s6-overlay.

This patch resolves the issue.

Submitted upstream: https://github.com/just-containers/s6-overlay/pull/335

---

diff -aur a/builder/overlay-rootfs/etc/s6/init/init-stage1 b/builder/overlay-rootfs/etc/s6/init/init-stage1
--- a/builder/overlay-rootfs/etc/s6/init/init-stage1
+++ b/builder/overlay-rootfs/etc/s6/init/init-stage1
@@ -58,7 +58,7 @@
 }
 { }

-backtick -n S6_CMD_ARG0 { printcontenv S6_CMD_ARG0 }
+backtick -D "" -n S6_CMD_ARG0 { printcontenv S6_CMD_ARG0 }
 importas -d "\n" -s -u S6_CMD_ARG0 S6_CMD_ARG0

 ##
diff -aur a/builder/overlay-rootfs/etc/s6/init/init-stage2 b/builder/overlay-rootfs/etc/s6/init/init-stage2
--- a/builder/overlay-rootfs/etc/s6/init/init-stage2
+++ b/builder/overlay-rootfs/etc/s6/init/init-stage2
@@ -35,7 +35,7 @@
         if { s6-echo -n -- "[s6-init] making user provided files available at /var/run/s6/etc..." }
         foreground
         {
-          backtick -n S6_RUNTIME_PROFILE { printcontenv S6_RUNTIME_PROFILE }
+          backtick -D "" -n S6_RUNTIME_PROFILE { printcontenv S6_RUNTIME_PROFILE }
           importas -u S6_RUNTIME_PROFILE S6_RUNTIME_PROFILE
           backtick -n S6_RUNTIME_PROFILE_SRC {
             ifte { s6-echo "/etc/cont-profile.d/${S6_RUNTIME_PROFILE}" } { s6-echo "/etc" }
